### DISCUSSION TITLE
Creating an application on moose

### DISCUSSION THREAD

**Discussion by Kenziekaia** [idaholab/moose#21164] [Q&A General] [answered]
When trying to create an an application on MOOSE it can not run jobs and submit output files. When performing `./Bash_CSV.sh` and then checking what the errors are via `VIM log_slurm.o<job number>`  however all previous tests have passed and worked. Every time I try to submit these jobs I get an error message that saying there is a segmentation fault and the user direction have been aborted. 

`Loading petsc/3.12.5
  Loading requirement: openmpi/gcc8/4.0.5 hypre/2.19.0 python37
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpiexec noticed that process rank 0 with PID 139056 on node cpu102 exited on signal 11 (Segmentation fault).`

If anyone has any suggestions it would be greatly appreciated. Thank you

**Comment by GiudGiud** [comment, marked as answer]
Please review the posting guidelines. There isnt nearly enough information to help you here. 

Also your petsc is really old.

**Reply by Kenziekaia** [reply]
Ok, I updated my discussion post with more info. Do I need to update my petsc?
Thanks, 
Kenzie

**Reply by GiudGiud** [reply]
I think that's one thing to try.
There isnt that much more information in your post.
What installation method did you follow? What compilers are you using?
Is this running well on interactive jobs/locally and crashing on submissions to the queue?

**Reply by Kenziekaia** [reply]
Ok thank you! I am sorry for the confusion/ lack of information it is my first time using MOOSE so I am trying to figuring out all of the terminology. 
**The installation method I used:** https://docs.google.com/document/d/1Hx5T7jtIdFMsvupkmKVPF4dMJ-s6DbYqh7XuOm4gsL0/edit?usp=sharing
**How I am making my application:** https://docs.google.com/document/d/1Hx5T7jtIdFMsvupkmKVPF4dMJ-s6DbYqh7XuOm4gsL0/edit?usp=sharing
I have been compiling using libmesh and PETSC
I believe the job is running well locally due to the job number being created but doesn't submit to the queue. But I am not sure.

**Reply by GiudGiud** [reply]
But running well locally I mean does it pass the tests? If you run ./run_tests in framework/test

**Reply by Kenziekaia** [reply]
I will try that after I finished updating PETSC and let you know

**Reply by GiudGiud** [reply]
Hello

So this looks fine but it doesnt explain why you have such an old petsc. My guess is that your environment is contaminated by other modules you are loading knowingly or not.
What does `module list` return?
You have shared the same document twice btw

Guillaume

**Reply by Kenziekaia** [reply]
Yeah I am not sure why I have such an old PETSC as we are using what is provided on BSU's borah. Here is the second document I meant to share which is how I am attempting to make an application: https://docs.google.com/document/d/1fuLiyAIhguw5ypJlsOeOStZ3S5PVF6z0WeVHaKWivPk/edit?usp=sharing
Thanks, 
Kenzie

**Reply by GiudGiud** [reply]
you need to remove that petsc from the environment.
what does `module list` return?

**Reply by Kenziekaia** [reply]
How could I remove the PETSC from the environment.
 **The `module list` command in my application returns:** 
`Currently Loaded Modulefiles:
 1) gcc/9.2.0   2) slurm/slurm/19.05.8   3) gcc/8.2.0   4) openmpi/gcc8/4.1.2   5) cmake/gcc8/3.18.0   6) valgrind/3.16.1   7) moose-dev-gcc  `

The `./run_tests` application passed and resulted in 
`test:kernels/simple_diffusion.test ........................................................................ OK
run_cr0.05_co0.05_T853/test:kernels/simple_diffusion.test ................................................. OK
run_cr0.05_co0.1_T853/test:kernels/simple_diffusion.test .................................................. OK
--------------------------------------------------------------------------------------------------------------
Ran 3 tests in 1.6 seconds. Average test time 0.5 seconds, maximum test time 0.5 seconds.
3 passed, 0 skipped, 0 pending, 0 failed`

**Reply by GiudGiud** [reply]
you have 2 gcc loaded, that is an issue as well. Use module unload to remove one of them. 

PETSC did not come from the modules it seems. It must be in your PATH or some other environment variable

**Reply by Kenziekaia** [reply]
I will try that!! When I just submitted my projects they went through so my fixes appeared to work.
